### PR TITLE
HDDS-6058. Bump aws-java-sdk to latest 1.12

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

Upgrade `aws-java-sdk` to 1.12.

https://issues.apache.org/jira/browse/HDDS-6058

## How was this patch tested?

CI
https://github.com/adoroszlai/hadoop-ozone/actions/runs/1526858610

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -237,7 +237,7 @@ xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xs
     <maven-site-plugin.version>3.9.1</maven-site-plugin.version>
     <checkstyle.version>8.29</checkstyle.version>
     <surefire.fork.timeout>1200</surefire.fork.timeout>
-    <aws-java-sdk.version>1.11.1034</aws-java-sdk.version>
+    <aws-java-sdk.version>1.12.122</aws-java-sdk.version>
```
Comment (ayushtkn) [inline pom.xml:240]: Guess we can upgrade to 124 as well right?
https://repo1.maven.org/maven2/com/amazonaws/aws-java-sdk/1.12.124/

Comment (adoroszlai) [reply pom.xml:240]: Sure.  Note: 124 is newer than the PR.

**Review by ayushtkn — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (ayushtkn) [reply pom.xml]: hmm, my bad. AWS folks are to be blamed they seems to be really very quick with releases. :-)

**Comment by adoroszlai** [comment]
Thanks @ayushtkn for the review.

**Pull request merged**